includes/sanitizers/class-amp-form-sanitizer.php

```diff
@@ -67,9 +67,21 @@ public function sanitize() {
 				$action_url = esc_url_raw( '//' . $_SERVER['HTTP_HOST'] . wp_unslash( $_SERVER['REQUEST_URI'] ) );
 			} else {
 				$action_url = $node->getAttribute( 'action' );
-				// Check if action_url is a relative path and add the host to it.
+
+				// Handle relative URLs.
 				if ( ! preg_match( '#^(https?:)?//#', $action_url ) ) {
-					$action_url = esc_url_raw( '//' . $_SERVER['HTTP_HOST'] . $action_url );
+					$schemeless_host = '//' . $_SERVER['HTTP_HOST'];
+					if ( '?' === $action_url[0] || '#' === $action_url[0] ) {
+						// For actions consisting of only a query or URL fragment, include the schemeless-host and the REQUEST URI of the current page.
+						$action_url = '//' . $_SERVER['HTTP_HOST'] . wp_unslash( $_SERVER['REQUEST_URI'] ) . $action_url;
+					} elseif ( '.' === $action_url[0] ) {
+						// For actions consisting of relative paths (e.g. '../'), prepend the schemeless-host and a trailing-slashed REQUEST URI.
+						$action_url = '//' . $_SERVER['HTTP_HOST'] . trailingslashit( wp_unslash( $_SERVER['REQUEST_URI'] ) ) . $action_url;
```

Comment:
I don't think the trailing slash should always be added. Taking the action `.foo` for example, depending on the server configuration `.foo` could be a file while `.foo/` could be a directory.

Reply:
Correct, but the trailing slash is being added after the path in order to append the relative segment. For example, if the user is currently on `/about/team` and there is a form that has an action of `../` the resulting `action` URL path needs to be `/about/team/../` not `/about/team../`. Does that make sense?

Reply:
Yes, thanks for the clarification.